src/tools/translate-to-fuzz.h

```diff
@@ -0,0 +1,1106 @@
+      }
+      case f64: {
+        return makeDeNaNOp(makeBinary({ pick(AddFloat64, SubFloat64, MulFloat64, DivFloat64, CopySignFloat64, MinFloat64, MaxFloat64), make(f64), make(f64) }));
+      }
+      default: WASM_UNREACHABLE();
+    }
+    WASM_UNREACHABLE();
+  }
+
+  Expression* makeSelect(const ThreeArgs& args) {
+    return builder.makeSelect(args.a, args.b, args.c);
+  }
+
+  Expression* makeSelect(WasmType type) {
+    return makeDeNaNOp(makeSelect({ make(i32), make(type), make(type) }));
+  }
+
+  Expression* makeSwitch(WasmType type) {
+    assert(type == unreachable);
+    if (breakableStack.empty()) return make(type);
+    // we need to find proper targets to break to; try a bunch
+    int tries = TRIES;
+    std::vector<Name> names;
+    WasmType valueType = unreachable;
+    while (tries-- > 0) {
+      auto* target = vectorPick(breakableStack);
+      auto name = getTargetName(target);
+      auto currValueType = getTargetType(target);
+      if (names.empty()) {
+        valueType = currValueType;
+      } else {
+        if (valueType != currValueType) {
+          continue; // all values must be the same
+        }
+      }
+      names.push_back(name);
+    }
+    if (names.size() < 2) {
+      // we failed to find enough
+      return make(type);
+    }
+    auto default_ = names.back();
+    names.pop_back();
+    auto temp1 = make(i32), temp2 = isConcreteWasmType(valueType) ? make(valueType) : nullptr;
+    return builder.makeSwitch(names, default_, temp1, temp2);
+  }
+
+  Expression* makeDrop(WasmType type) {
+    return builder.makeDrop(make(type == unreachable ? type : getConcreteType()));
+  }
+
+  Expression* makeReturn(WasmType type) {
+    return builder.makeReturn(isConcreteWasmType(func->result) ? make(func->result) : nullptr);
+  }
+
+  Expression* makeNop(WasmType type) {
+    assert(type == none);
+    return builder.makeNop();
+  }
+
+  Expression* makeUnreachable(WasmType type) {
+    assert(type == unreachable);
+    return builder.makeUnreachable();
+  }
+
+  // special getters
+
+  WasmType getType() {
+    switch (upTo(6)) {
+      case 0: return i32;
+      case 1: return i64;
+      case 2: return f32;
+      case 3: return f64;
+      case 4: return none;
+      case 5: return unreachable;
+    }
+    WASM_UNREACHABLE();
+  }
+
+  WasmType getReachableType() {
+    switch (upTo(5)) {
+      case 0: return i32;
+      case 1: return i64;
+      case 2: return f32;
+      case 3: return f64;
+      case 4: return none;
+    }
+    WASM_UNREACHABLE();
+  }
+
+  WasmType getConcreteType() {
+    switch (upTo(4)) {
+      case 0: return i32;
+      case 1: return i64;
+      case 2: return f32;
+      case 3: return f64;
+    }
+    WASM_UNREACHABLE();
+  }
+
+  // statistical distributions
+
+  // 0 to the limit, logarithmic scale
+  Index logify(Index x) {
+    return std::floor(std::log(std::max(Index(1) + x, Index(1))));
+  }
+
+  Index upTo(Index x) {
```

Comment:
It's also like a lot of `rand` calls (like http://introcs.cs.princeton.edu/java/12types/RandomInt.java.html). These all make sense once you know what they're doing, my issue with the name `upTo` is just that to me it doesn't say it's one of those kinds of functions, and "up to" in particular sounds distinct from "up to but not including".
In my head using `get` in the name ties it to the `get` methods already in this class, which are random, implying that this is also random, and therefore an exclusive range.

But, no I don't feel particularly strongly about it.